Falons branch

FalonMadeIt opened the pull request:
This is a small dictionary and a filter. A larger dictionary can be added.

FalonMadeIt committed:
added dictionary and word filter

FalonMadeIt committed:
adding js for filter

marcusp619 commented:
This looks good for the initial start!

sarony commented on scripts.js:
```diff
@@ -0,0 +1,38 @@
+ "could","people","my","than","first","water","been","call",
+ "who","oil","its","now","find","long","down","day","did","get",
+ "come","made","may","part"
+];
```
What do you think of moving this array into a file instead? That way, when we get a bigger dictionary, we can just replace the filename so that the swap will be easier.

coryunderwood replied:
It makes sense, it could be loaded via it's own script file, via a JSON object, or refactored to make a AJAX call to a DB and return a word.   Many options available.

marcusp619 replied:
Here is Link to where you can download The English Open Word List. 

http://dreamsteep.com/projects/53-the-english-open-word-list-eowl.html

sarony commented on scripts.js:
```diff
@@ -0,0 +1,38 @@
+(function () {
+  console.log('hey');
```
I'm wondering what this line of code does. I'm thinking it was just to make sure your code was working. Do we still need it?

coryunderwood replied:
Console.log is normally used for testing / development.  Likely you do not want to write to the console unless you have very specific reasons to in the finished code.

sarony commented:
@falont23 I made a few comments and had a few questions. Let me know when you get a chance to review and I'd love to take another look.

sarony commented on scripts.js:
```diff
@@ -0,0 +1,38 @@
+(function () {
```
@simonxca was wondering why this code is wrapped in a function and if this is the norm. @coryunderwood want to comment on this?

coryunderwood replied:
This makes it a self executing block of code, and prevents anything inside of it from being in the global namespace.  Access to the variables and functions inside the closure must be made public through some sort of namespace interaction, or else nothing can 'talk' to the code inside the block.  It's pretty much a scoping issue, which causes similar behavior to Java's Public / Private methods inside of a larger class.

Use of this form is normal if you want to prevent your code from polluting the global space, or if you needed it to self execute on parse, rather than calling the function via some other block of code.

FalonMadeIt committed:
Update scripts.js

I deleted the encapsulating "function" now there should be no scope issue. Please check and let me know!

dsberger committed:
deletes some console.log statements

FalonMadeIt committed:
Update scripts.js

Deleting unused function.

FalonMadeIt committed:
adding dictionary to existing script file
